### DISCUSSION TITLE
LEDを取り付けた後に反応しなくなった。

### DISCUSSION THREAD

**Discussion by homaji** [Yowkees/keyball#373] [Q&A]
似たような質問が多数ありそうでしたが、微妙に違う感じもしたのでこちらで質問させてください。

Keyball39をLED無しで組み立てて、問題なく動作していました。
1週間ほど利用したのち、LEDの追加を行ったところ、USBそのものを認識しない状態になりました。
どのように検討していてばいいのか、アドバイスをいただけないでしょうか。

行ったこと
１．右側にトラックボールを配置し、LED無しで組み立て
２．右と左のどちらも公式のファームを書き込み済。
３．LEDを追加、右・左のどちらのUSBポートにつないでもLEDもOLEDも点灯しない。
４．TSRケーブルを外し、左側単体だとOLEDもLEDも点灯、入力も問題なし。
５．右側単体では、点灯もなく、パソコンから認識もできない。ファームウェアの書き込み先にリセットしてもでてこない。
６．LEDの通電は１番のDINから２２番のDOUTで通電は確認。
７．怪しいつき方になっていたLEDは交換して、再度はんだ付
この状態でも認識しません。

自分の行ったなんらかの作業が悪さしただろうとは思うのですが、どこを順番に見るべきか教えてください。
#363 を見て、まずはProMicroを確認しようと思うのですが、コンスルーでない場合にProMicroを外すのはどうすればいいでしょうか？
また、その次に確認すべき箇所はどういうところがありますか？

![PXL_20231001_083000235](https://github.com/Yowkees/keyball/assets/1880055/ca5f71a9-2092-4bce-ad40-a3650e77cab6)

**Comment by homaji** [comment]
追記です。
ProMicroって引っ張ればとれるんですね。
ProMicro単体で接続すると、ランプがつくことは確認できました。

**Comment by homaji** [comment]
さらに確認
左と右のProMicroを入れ替えて接続
左の基盤＋右のProMicro　：起動
右の基盤＋左のProMicro　：起動せず

右の基盤のどこかがおかしいと判断しました。

一度全部のLEDを外してみます。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ご連絡が遅くなり、いろいろご対応いただきありがとうございます。

LEDはショートモードで壊れることがあります。
一応、全LEDを電源から切り離して確認する方法はありますが、確認後は結局１個ずつ取り外し、壊れているものを探すことになりますので、思い切ってすべてやり直しても良いと思います。
半田ごてが２本あると、ちょんちょんと挟みながら簡単に除去できます。

お手数ですがLEDの故障確認よろしくお願いいたします。

**Reply by homaji** [reply]
ご連絡ありがとうございます。
平日にはなかなか時間がとれませんが、アドバイスいただいた内容で確認してみます。
技術持ってる方にコメントしていただけると、作業に取り掛かる際の不安感が少なくなるので非常に助かります。

**Comment by homaji** [comment]
@Yowkees 

LEDをすべて取り外したところ、元通り動作するようになりました。
取り外す際に失敗して、基盤の一部が壊れたので、LEDの搭載は左手だけで我慢します😂

最後にお尋ねしたいのですが、LEDを搭載してキーが反応しなくなったのはなぜでしょうか？
LEDの半田に失敗しても途中まで点灯するという説明が多かったので、LED部分の回路だけ失敗してもキーボードは問題ないのかなと考えていました。
キーボード全体を巻き込んで動かなくなるというのは、下記のようなものが想定されるのでしょうか？
１．半田の際にLEDが壊れて、内部でショートして、全体へ影響する
２．LEDの半田に失敗して隣とショートしたので、全体へ影響する

お時間のあるときにご回答いただけると嬉しいです。

**Comment by Yowkees** [comment]
ひとまず復活できたようで良かったです。
LEDのVCCとGNDがショートした状態で壊れてしまうことがあり、この場合には電源のプラスとマイナスがショートしている状態ですので、キーボード全体が動作しなくなってしまいます。
小手先温度が３００℃以上だったり、LEDが湿気っていたりすると壊れやすいように思います。

以上、よろしくお願いいたします。
